tasks/maybe_connect_dev.js

```diff
@@ -1,39 +1,52 @@
 module.exports = function (grunt) {
-  grunt.registerTask('maybe_start_server', function () {
-    var http = require('http');
+  var maybeStartServer = function (options) {
+    return function () {
+      var http = require('http');
 
-    var req = http.request({
-      method: 'HEAD',
-      path: '/',
-      host: 'localhost',
-      port: 8000
-    });
+      var req = http.request({
+        method: 'HEAD',
+        path: '/',
+        host: 'localhost',
+        port: options.port
+      });
 
-    function onResponse(res) {
-      if (res.headers.pong === 'Kibana 4 Dev Server') {
-        grunt.log.writeln('server already started');
-      } else {
-        grunt.log.error('another server is already running at localhost:8000!');
+      function onResponse(res) {
+        if (res.statusCode === 200) {
+          grunt.log.writeln('server already started');
+        } else {
+          grunt.log.error('another server is already running at localhost:8000!');
+        }
+        done(res);
       }
-      done(res);
-    }
 
-    function onError() {
-      grunt.task.run(['server']);
-      done();
-    }
+      function onError() {
+        grunt.config.set(options.name, true);
+        grunt.task.run(options.tasks);
+        done();
+      }
 
-    var done = (function (cb) {
-      return function (res) {
-        req.removeListener('error', onError);
-        req.removeListener('response', onResponse);
-        if (res) res.socket.destroy();
-        cb();
-      };
-    })(this.async());
+      var done = (function (cb) {
+        return function (res) {
+          req.removeListener('error', onError);
+          req.removeListener('response', onResponse);
+          if (res) res.socket.destroy();
+          cb();
+        };
+      })(this.async());
 
-    req.on('error', onError);
-    req.on('response', onResponse);
-    req.end();
-  });
-};
\ No newline at end of file
+      req.on('error', onError);
+      req.on('response', onResponse);
+      req.end();
+    };
+  };
+  grunt.registerTask('maybe_start_server', maybeStartServer({
+    name: 'connect-server',
+    port: 8000,
+    tasks: ['server']
+  }));
+  grunt.registerTask('maybe_start_kibana', maybeStartServer({
```

Comment:
I'm confused by this, is the connect server around simply to support the previous dev_server behavior?

It makes sense to me that there would be one server, possibly with different behavior in dev mode (serving tests for instance).